Sources/SwiftUIKit/Views/Map.swift

```diff
@@ -0,0 +1,486 @@
+//
+//  Map.swift
+//  SwiftUIKit
+//
+//  Created by Oskar on 12/04/2020.
+//
+
+import Foundation
+import MapKit
+
+public struct MapPoint {
+  let latitude: Double
+  let longitude: Double
+  let title: String
+  let subtitle: String
+}
+
+public class Map: MKMapView {
+  
+  fileprivate var initialCoordinates: CLLocationCoordinate2D
+  
+  fileprivate lazy var onFinishLoadingHandler: ((MKMapView) -> ())? = nil
+  
+  fileprivate lazy var afterRegionChangeHandler: ((MKMapView) -> ())? = nil
+  
+  fileprivate lazy var beforeRegionChangeHandler: ((MKMapView) -> ())? = nil
+  
+  fileprivate lazy var annotationViewConfigurationHandler: ((MKAnnotationView?, MKAnnotation) -> (MKAnnotationView?))? = nil
+  
+  fileprivate lazy var onAccessoryTapHandler: ((MKMapView, MKAnnotationView, UIControl) -> ())? = nil
+  
+  fileprivate lazy var onAnnotationViewStateChangeHandler: ((MKMapView, MKAnnotationView, MKAnnotationView.DragState, MKAnnotationView.DragState) -> ())? = nil
+  
+  fileprivate lazy var onAnnotationSelectHandler: ((MKMapView, MKAnnotationView) -> ())? = nil
+  
+  fileprivate lazy var onAnnotationDeselectHandler: ((MKMapView, MKAnnotationView) -> ())? = nil
+  
+  fileprivate lazy var annotationViewIdentifier: String? = nil
+  
+  public init(lat latitude: Double,
+              lon longitude: Double,
+              points: (() -> [MapPoint])? = nil) {
+    let coordinates = CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
+    
+    initialCoordinates = coordinates
+    
+    super.init(frame: .zero)
+    
+    let span = MKCoordinateSpan(latitudeDelta: region.span.latitudeDelta / 2,
+                                longitudeDelta: region.span.longitudeDelta / 2)
+    
+    let region = MKCoordinateRegion(center: coordinates, span: span)
+    setRegion(region, animated: true)
+    
+    if let points = points {
+      add(points: points())
+    }
+    
+    self.delegate = self
+  }
+  
+  convenience init(region: MKCoordinateRegion,
+                   points: (() -> [MapPoint])? = nil) {
+    self.init(lat: region.center.latitude,
+              lon: region.center.longitude,
+              points: points)
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+}
+
+// MARK: - Accessing Map Properties
+extension Map {
```

Comment:
if you make `extension` ... `public extension` everything inside it automatically inherits `public`

Reply:
[Commit](https://github.com/0xLeif/SwiftUIKit/pull/132/commits/8d7448ad24597d8fc0888569c52516a444df2b18)